packages/doenetml-prototype/src/doenet-applet.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useRef } from "react";
+//@ts-ignore
+import { prng_alea } from "esm-seedrandom";
+import { PageViewer } from "./viewer/page-viewer";
+import { Provider } from "react-redux";
+import { store } from "./state/store";
+let rngClass = prng_alea;
+
+export type DoenetMLFlags = {
+    showCorrectness: boolean;
+    readOnly: boolean;
+    solutionDisplayMode: string;
+    showFeedback: boolean;
+    showHints: boolean;
+    allowLoadState: boolean;
+    allowSaveState: boolean;
+    allowLocalState: boolean;
+    allowSaveSubmissions: boolean;
+    allowSaveEvents: boolean;
+    autoSubmit: boolean;
+};
+
+type DoenetMLFlagsSubset = Partial<DoenetMLFlags>;
+
+const defaultFlags: DoenetMLFlags = {
+    showCorrectness: true,
+    readOnly: false,
+    solutionDisplayMode: "button",
+    showFeedback: true,
+    showHints: true,
+    allowLoadState: false,
+    allowSaveState: false,
+    allowLocalState: false,
+    allowSaveSubmissions: false,
+    allowSaveEvents: false,
+    autoSubmit: false,
+};
+
+export function DoenetApplet({
```

Comment:
I don't think of DoenetML documents as applets, so I'd change the wording. Other possible words are documents or activities, though avoiding any such term is OK. Plus, Doenet and DoenetML are distinct, and I think everything here is DoenetML.

Reply:
Sure, I can change it back if you like.

My thinking was the following: we have `PageViewer` which renders things and assumes the state, etc. has already been defined. It just uses callbacks. Then we have `DoenetApplet` which is the "root" of a self-contained doenet instance. It might split the source into multiple pages and call several instances of `PageViewer` or it might render everything together. If you have multiple instances of `DoenetApplet`, they cannot communicate.

I felt like `DoenetML` was a little confusing becase
  1. You're not just inserting some doenet code, you're creating an isolated context.
  2. The `ML` part stands for Markup Language; I know you're distinguishing the Doenet server/LMS from the Doenet render/interface, but I think `ML` is a strange way to distinguish this...